include/core/physics_solver.h

```diff
@@ -25,11 +27,30 @@ template <typename VectorType>
 class PhysicsSolver
 {
 public:
+  /**
+   * @brief Constructor for the non-linear physics.
+   *
+   * @param[in] param Non-linear solver parameters as specified in the
+   * simulation parameter file.
+   *
+   */
   PhysicsSolver(const Parameters::NonLinearSolver non_linear_solver_parameters);
 
+
+  /**
+   * @brief Constructor for the linear physics.
+   *
```

Comment:
```suggestion
   * @brief Constructor for the linear physics. Since the Physics is linear, a LinearSolutionStrategy is automatically generated as a PhysicsSolverStrategy.
   *
```

I find it worth it to be extra verbose in documentation.